class Name(s)
class Character(name: my Name)

fn name(c: Character) -> shared{c} Name {
    c.name.share 
    #! RUN ERROR expected a `our` value, got a `shared` value
    #
    # This error message isn't great and points to some holes in our terminology.
    #
    # What is happening here:
    # * we are given a `Character` with full permissions
    # * we execute `c.name.share`, which gets a shared lease on the name
    # * but `shared{c}` expects shared *ownership*, since we had ownership coming in
}

c1 = Character(Name("Achilles"))
n_given = name(c1.give)
print(n_given).await
